src/main/scala/io/phdata/retirementage/SparkDriver.scala

```diff
@@ -52,7 +57,11 @@ object SparkDriver extends LazyLogging {
             filter.noAction()
           } else {
             // table has no hold and needs to be filtered, write a new dataset without expired records
-            filter.doFilter(computeCountsFlag, dryRun = false)
+            if (filter.isInstanceOf[KuduTableFilter]) {
+              filter.doKuduFilter(computeCountsFlag, dryRun = false)
```

Comment:
there should just be 'filter' method and the kudu implementation should override 'filter'